controllers/configmap_controller.go

```diff
@@ -77,8 +89,20 @@ func (r *ConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		log.Error(err, "unable to list configmaps")
 		return ctrl.Result{}, nil
 	}
+	log.Info("List: got", "itemCount", len(configMapList.Items))
+	found := false
 	for _, cm := range configMapList.Items {
-		log.Info("List: got", "clusterName", logicalcluster.From(&cm).String(), "namespace", cm.Namespace, "name", cm.Name)
+		if "" != logicalcluster.From(&cm).String() {
+			log.Info("List: got", "clusterName", logicalcluster.From(&cm).String(), "namespace", cm.Namespace, "name", cm.Name)
+		} else {
+			if cm.Name == configMap.Name && cm.Namespace == configMap.Namespace {
+				if found {
+					return ctrl.Result{}, fmt.Errorf("there should be listed only one configmap with the given name '%s' for the given namespace '%s' when the clusterName is not available", cm.Name, cm.Namespace)
+				}
+				found = true
+				log.Info("Found in listed configmaps", "namespace", cm.Namespace, "name", cm.Name)
```

Comment:
I'd move this back out (just inside the else block) and change it back to the `List: got...` log message. There can be multiple configmaps in a cluster across all namespaces, and this is supposed to show all of them.

Reply:
Yes, but imagine what it looks like when it's "connected" to a normal OpenShift cluster - there are approx. 550 ConfigMaps in the cluster and all of them are printed every time when the leader election updates its own configmap. This makes the whole log unreadable. 
That's why I change it so it prints the count of all CMs found either in the cluster or in the workspace, plus:
* when it's running against kcp cluster (clusterName is set) then it shows all of them (as you requested)
* when it's running against a normal cluster, then it only finds the right ConfigMap which triggered the reconcile, and prints only that one